pkg/controller/networkpolicy/status_controller.go

```diff
@@ -283,38 +293,60 @@ func (c *StatusController) syncHandler(key string) error {
 		return c.npControlInterface.UpdateAntreaClusterNetworkPolicyStatus(internalNP.SourceRef.Name, status)
 	}
 
+	conditions := GenerateNetworkPolicyCondition(internalNP.SyncError)
 	// It means the NetworkPolicy has been processed, and marked as unrealizable. It will enter unrealizable phase
 	// instead of being further realized. Antrea-agents will not process further.
 	if internalNP.SyncError != nil {
-		return updateStatus(crdv1alpha1.NetworkPolicyPending, 0, 0)
+		return updateStatus(crdv1alpha1.NetworkPolicyPending, 0, 0, conditions)
 	}
 
 	// It means the NetworkPolicy hasn't been processed once. Set it to Pending to differentiate from NetworkPolicies
 	// that spans 0 Node.
 	if internalNP.SpanMeta.NodeNames == nil {
-		return updateStatus(crdv1alpha1.NetworkPolicyPending, 0, 0)
+		return updateStatus(crdv1alpha1.NetworkPolicyPending, 0, 0, conditions)
 	}
 
 	desiredNodes := len(internalNP.SpanMeta.NodeNames)
 	currentNodes := 0
 	statuses := c.getNodeStatuses(key)
+	failedRealizedNodes := make([]string, 0)
 	for _, status := range statuses {
 		// The node is no longer in the span of this policy, delete its status.
 		if !internalNP.NodeNames.Has(status.NodeName) {
 			c.deleteNodeStatus(key, status.NodeName)
 			continue
 		}
 		if status.Generation == internalNP.Generation {
-			currentNodes += 1
+			// To be compatible with the previous versions, it means the NetworkPolicy realization is succeeded
```

Comment:
I'm not quite sure I understand this comment

Reply:
I guess: "If status.RealizationFailure is not set, it means NetworkPolicy realization succeeded on the Node. This is to be compatible with the previous implementation when status.RealizationFailure was not added."

Reply:
Jianjun is correct. Since this field is newly added, it means NetworkPolicy realization succeeded in the previous implementations when this field (status.RealizationFailure) is not set. 

Maybe I should remove this comment, as I changes the field type from pointer to bool, and the default value is false which is used when controller works with agents in the previous versions.